pages/docs/concepts/asyncapi-document/resuable-parts.md

```diff
@@ -0,0 +1,130 @@
+
+```mermaid
+graph LR
+style A fill:#47BCEE, stroke:#333, stroke-width:2px;
+style C fill:#47BCEE, stroke:#333, stroke-width:2px;
+style D fill:#47BCEE, stroke:#333, stroke-width:2px;
+style B fill:#47BCEE, stroke:#333, stroke-width:2px;
+
+A[MyMessageSchema]
+B[components]
+C[myChannel]
+D[$ref keyword]
+
+C -->|publishes| D
+D -->|references| B
+B -->|references| A
+```
+
+In the example below, you define a component called MyMessageSchema under the schemas section to describe the structure of a message. Under the publish operation of myChannel, you reference the MyMessageSchema component using the $ref keyword.
+
+```yaml
+channels:
+  myChannel:
+    publish:
+      message:
+        $ref: '#/components/schemas/MyMessageSchema'
+components:
+  schemas:
+    MyMessageSchema:
+      type: object
+      properties:
+        message:
+          type: string
+```
+
+## Another local document
+
+You can use the ref keyword to reference another local document. Ensure the path to the local file is correct and accessible from your main AsyncAPI document.
+
+The image below defines the process of referencing another local document.
+
+```mermaid
+graph LR
+style A fill:#47BCEE, stroke:#333, stroke-width:2px;
+style C fill:#47BCEE, stroke:#333, stroke-width:2px;
+style D fill:#47BCEE, stroke:#333, stroke-width:2px;
+style B fill:#47BCEE, stroke:#333, stroke-width:2px;
+
+A[Main AsyncAPI Document]
+B[message-schema.yaml]
+C[components/schemas]
+D[MyMessageSchema]
+
+A -->|references| B
+B -->|contains| C
+C -->|defines| D
+```
+
+In the example below, you reference the component from the message-schema.yaml file.
+
+```yaml
+##### ./message-schema.yaml
+
+components:
+  messages:
+    UserSignup:
+      name: UserSignup
+      title: User signup
+      summary: Action to sign a user up.
+      description: A longer description
+      contentType: application/json
+      payload: null
+```
+
+```yaml
+##### ./asyncapi.B.yaml
+
+channels:
+  user/signedup:
+    publish:
+      message:
+        $ref: ./asyncapi.A.yaml#/components/messages/UserSignup
+```
+
+## External URL
+
+You can use the ref keyword to reference an external URL. Ensure the external URL should provide the referenced component in a compatible format, such as YAML or JSON.
+
+The image below defines the process of referencing an external URL.
+
+```mermaid
+graph LR
+style A fill:#47BCEE, stroke:#333, stroke-width:2px;
+style C fill:#47BCEE, stroke:#333, stroke-width:2px;
+style D fill:#47BCEE, stroke:#333, stroke-width:2px;
+style B fill:#47BCEE, stroke:#333, stroke-width:2px;
+
+A[AsyncAPI Document]
+B[https://example.com/my-component.yaml]
+C[components/schemas]
+D[MyComponent]
+
+A -->|references| B
+B -->|contains| C
+C -->|defines| D
+```
+
+In the example below, you reference the component from an external URL. The $ref value specifies the full URL to the external resource and the component's location.
+
+```yaml
+##### ./asyncapi.A.yaml
```

Comment:
Same question as line 88... Why the need for `#####` included in this line? Seems like a mistake. 😄

Reply:
We need to indicate file names, since we are using a couple of files.